Add support for flipping BoxContainer children

timothyqiu opened the pull request:
Closes https://github.com/godotengine/godot-proposals/issues/804

The actual flipping logic is already implemented for RTL support, so this PR just adds an extra toggle.

Yeah, the same thing can be achieved by moving newly added nodes to the top:

```gdscript
add_child(node)
move_child(node, 0)
```

But it's a hack tying logical order with physical order.

timothyqiu committed:
Add support for flipping BoxContainer children

groud commented:
I don't mind much, but the linked proposal does not have a lot of support.
And, well, my comments [there](https://github.com/godotengine/godot/pull/35791)  do still apply: it's simple to work around and the use case isn't really clear IMO.

timothyqiu commented:
If the use case is significant enough, it's probably already covered. If it's not that significant, then a simple use case like that can always be worked around with a few lines of code. And then we go in a circle :P

Anyway, we've already implemented such option for `FlowContainer` in #74195.

Updated the property's name to match that.

timothyqiu commented:
Closing. It's too exhausting to come back from time to time to check if a PR from half a year ago needs to be rebased.
